File: server/src/Routes/EmployeeRoutes.js
import express from "express";
import JWT from "../Middlewares/jsonwebtoken.js";
import EmployeeController from "../Controllers/employeeController.js";

const EmployeeRoutes = express.Router();

EmployeeRoutes.get(
  "/get-one/:employeeId",
  JWT.validateAccessToken,
  EmployeeController.getEmployee
);
EmployeeRoutes.post(
  "/create",
  JWT.validateAccessToken,
  EmployeeController.createEmployee
);
EmployeeRoutes.get(
  "/get-all",
  JWT.validateAccessToken,
  EmployeeController.getAllEmployees
);
EmployeeRoutes.put(
  "/update/:employeeId",
  JWT.validateAccessToken,
  EmployeeController.updateEmployee
);
EmployeeRoutes.delete("/delete/:employeeId", JWT.validateAccessToken, EmployeeController.deleteEmployee)

export default EmployeeRoutes;

File: server/src/routes/employeeRoutes.go
package routes

import (
	"controll/src/controllers"

	"github.com/gin-gonic/gin"
)

func EmployeeRoutes(router *gin.RouterGroup) {
	employee := router.Group("/api/employee")
	{
		employee.GET("/get-one/:employeeId", controllers.GetEmployee())
		employee.POST("/create", controllers.CreateEmployee())
		employee.GET("/get-all", controllers.GetAllEmployees())
		employee.PUT("/update/:employeeId", controllers.UpdateEmployee())
		employee.DELETE("/delete/:employeeId", controllers.DeleteEmployee())
	}
}
